src/Microsoft.IdentityModel.Tokens/Validation/Results/Details/LifetimeValidationError.cs

```diff
@@ -9,37 +9,21 @@ namespace Microsoft.IdentityModel.Tokens
 {
     internal class LifetimeValidationError : ValidationError
     {
-        protected DateTime _notBefore;
-        protected DateTime _expires;
-
-        public LifetimeValidationError(
-            MessageDetail messageDetail,
-            Type exceptionType,
-            StackFrame stackFrame)
-            : base(messageDetail, ValidationFailureType.LifetimeValidationFailed, exceptionType, stackFrame)
-        {
-        }
-
         public LifetimeValidationError(
             MessageDetail messageDetail,
             Type exceptionType,
             StackFrame stackFrame,
-            DateTime notBefore,
-            DateTime expires)
-            : base(messageDetail, ValidationFailureType.LifetimeValidationFailed, exceptionType, stackFrame)
-        {
-            _notBefore = notBefore;
-            _expires = expires;
-        }
+            DateTime? notBefore,
+            DateTime? expires,
+            ValidationFailureType? validationFailureType = null,
+            Exception? innerException = null)
 
-        public LifetimeValidationError(
-            MessageDetail messageDetail,
-            Type exceptionType,
-            StackFrame stackFrame,
-            DateTime expires)
-            : base(messageDetail, ValidationFailureType.LifetimeValidationFailed, exceptionType, stackFrame)
+            : base(messageDetail, exceptionType, stackFrame, validationFailureType ?? ValidationFailureType.LifetimeValidationFailed, innerException)
         {
-            _expires = expires;
+            if (notBefore.HasValue)
```

Comment:
If type is nullable, DateTime? then just set the property.

Reply:
Updated.